include/swift/Sema/TypeCheckRequest.h

```diff
@@ -114,7 +114,42 @@ class TypeCheckRequest {
   }
 
 #include "swift/Sema/TypeCheckRequestPayloads.def"
-  
+
+  TypeCheckRequest(const TypeCheckRequest &T) { *this = T; }
+
+  TypeCheckRequest& operator=(const TypeCheckRequest &T) {
```

Comment:
No, it wouldn't. I tried that earlier and clang rejects it, I think correctly.
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2008/n2544.pdf and other papers talk about this.